src/vm/types/signatures.rs

```diff
@@ -64,7 +64,7 @@ impl AssetIdentifier {
 
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
 pub struct TupleTypeSignature {
-    type_map: BTreeMap<ClarityName, TypeSignature>,
+    pub type_map: BTreeMap<ClarityName, TypeSignature>,
```

Comment:
Does this need to be `pub`?

Reply:
Yeah when doing the merge operation, I'm draining, in order to avoid cloning. Do you prefer a method returning a mutable ref? or cloning?

Reply:
Sorry for the slow response -- it looks like the draining/mutation is all happening in `shallow_merge`,  which is defined in this same module, so the `pub` shouldn't be necessary.

Reply:
Indeed, got confused by `pub` behavior, thanks. 
Addressed with https://github.com/blockstack/stacks-blockchain/pull/2117/commits/65706f366960b80874843ec3bf07e8dd5a94ba91

Reply:
Perhaps a reference counted pointer could be of use here:
https://doc.rust-lang.org/book/ch15-04-rc.html